qiskit/algorithms/gradients/finite_diff_estimator_gradient.py

```diff
@@ -0,0 +1,96 @@
+from qiskit.quantum_info.operators.base_operator import BaseOperator
+
+from .base_estimator_gradient import BaseEstimatorGradient
+from .estimator_gradient_result import EstimatorGradientResult
+
+
+class FiniteDiffEstimatorGradient(BaseEstimatorGradient):
+    """
+    Compute the gradients of the expectation values by finite difference method.
+    """
+
+    def __init__(self, estimator: BaseEstimator, epsilon: float = 1e-2, **run_options):
+        """
+        Args:
+            estimator: The estimator used to compute the gradients.
+            epsilon: The offset size for the finite difference gradients.
+            run_options: Backend runtime options used for circuit execution. The order of priority is:
+                run_options in `run` method > gradient's default run_options > primitive's default
+                setting. Higher priority setting overrides lower priority setting.
+        """
+        self._epsilon = epsilon
+        self._base_parameter_values_dict = {}
+        super().__init__(estimator, **run_options)
+
+    def _run(
+        self,
+        circuits: Sequence[QuantumCircuit],
+        observables: Sequence[BaseOperator | PauliSumOp],
+        parameter_values: Sequence[Sequence[float]],
+        parameters: Sequence[Sequence[Parameter] | None] | None = None,
+        **run_options,
+    ) -> EstimatorGradientResult:
+        """Compute the estimator gradients on the given circuits."""
+        # if parameters is none, all parameters in each circuit are differentiated.
+        if parameters is None:
+            parameters = [None for _ in range(len(circuits))]
+
+        jobs, result_indices_all = [], []
+        for circuit, observable, parameter_values_, parameters_ in zip(
+            circuits, observables, parameter_values, parameters
+        ):
+            # indices of parameters to be differentiated
+            if parameters_ is None:
+                indices = list(range(circuit.num_parameters))
+            else:
+                indices = [circuit.parameters.data.index(p) for p in parameters_]
+            result_indices_all.append(indices)
+
+            offset = np.identity(circuit.num_parameters)[indices, :]
+            plus = parameter_values_ + self._epsilon * offset
+            minus = parameter_values_ - self._epsilon * offset
+            n = 2 * len(indices)
+
+            job = self._estimator.run(
+                [circuit] * n, [observable] * n, plus.tolist() + minus.tolist(), **run_options
+            )
+            jobs.append(job)
+
+        # combine the results
+        results = [job.result() for job in jobs]
+        gradients, metadata_ = [], []
+        for i, result in enumerate(results):
+            n = len(result.values) // 2  # is always a multiple of 2
+            gradient_ = (result.values[:n] - result.values[n:]) / (2 * self._epsilon)
+            indices = result_indices_all[i]
+            gradient = np.zeros(circuits[i].num_parameters)
```

Comment:
If my understanding is correct, when the partial derivative is computed, the un-computed values are zero. There is a difference in meaning between zero and un-computed.

I suggest the possible options are to return an array of sizes computed, or to include `np.nan`?

Reply:
+1, but if so, we should store the information about the parameter to be derivatived. (metadata?)

Reply:
I like the idea to store parameters to metedata.
(Julien's example) E.g. if I have a circuit with parameters x0 x1 x2 ... x9 and I specify gradient.run([circuit], [obs], [values], [x1]), the size of the gradient is 1 and `metadata['parameters']=[x1]`